### DISCUSSION TITLE
Lint/RedundantDirGlobSort is not redundant

### DISCUSSION THREAD

**Discussion by pjscopeland** [rubocop/rubocop#12917] [Maintenance]
`Lint/RedundantDirGlobSort` wants to remove `.sort` but it isn't actually redundant.
Ruby 3.2.2
[Project-specific information redacted]

`Dir[glob] #=>` | `Dir[glob].sort #=>`
---|---
`["spec/controllers/█/█.rb",`<br>` "spec/controllers/█.rb",`<br>` "spec/filters/█.rb",`<br>` "spec/models/reports/█.rb",`<br>` "spec/models/█.rb",`<br>` "spec/acceptance/█.rb",`<br>` "spec/controllers/api/v1/█.rb",`<br>` "spec/controllers/█.rb",`<br>` "spec/filters/api/v1/█.rb",`<br>` "spec/filters/█.rb",`<br>` "spec/models/█.rb",`<br>` "spec/models/█.rb",`<br>` "spec/models/█.rb",`<br>` "spec/models/query_objects/█.rb",`<br>` "spec/serializers/api/v1/█.rb",`<br>` "spec/services/█.rb",`<br>` "spec/services/exception_checks/█.rb"]` | `["spec/acceptance/█.rb",`<br>` "spec/controllers/api/v1/█.rb",`<br>` "spec/controllers/█.rb",`<br>` "spec/controllers/█/█.rb",`<br>` "spec/controllers/█.rb",`<br>` "spec/filters/api/v1/█.rb",`<br>` "spec/filters/█.rb",`<br>` "spec/filters/█.rb",`<br>` "spec/models/█.rb",`<br>` "spec/models/█.rb",`<br>` "spec/models/█.rb",`<br>` "spec/models/query_objects/█.rb",`<br>` "spec/models/reports/█.rb",`<br>` "spec/models/█.rb",`<br>` "spec/serializers/api/v1/█.rb",`<br>` "spec/services/█.rb",`<br>` "spec/services/exception_checks/█.rb"]`

**Comment by Earlopain** [comment]
This cops autocorrect is marked as unsafe specifically for the case your are writing about here: https://docs.rubocop.org/rubocop/cops_lint.html#safety-22

The cop was initially marked as safe and I don't think it would have gone through if the safety issue was know about from the start.
